base/src/main/java/com/mashup/base/BaseRecyclerView.kt

```diff
@@ -0,0 +1,58 @@
+package com.mashup.base
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.recyclerview.widget.RecyclerView
+
+abstract class BaseRecyclerView {
+
+    abstract class Adapter<ITEM : Any, B : ViewDataBinding>(
+        @LayoutRes private val layoutRes: Int,
+        private val bindingVariableId: Int? = null
+    ) : RecyclerView.Adapter<ViewHolder<B>>() {
+
+        private val items = mutableListOf<ITEM>()
+
+        fun replaceAll(newItems: List<ITEM>?) {
+            newItems?.let {
+                items.run {
+                    clear()
+                    addAll(it)
+                }
+            }
+        }
+
+        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder<B> =
+            object : ViewHolder<B>(
+                parent = parent,
+                layoutRes = layoutRes,
+                bindingVariableId = bindingVariableId
```

Comment:
이 부분에서 Databinding 된 객체를 넘기는건 어떻게 생각하세요? 아래처럼 넘기게 되면 RecyclerViewHolder 생성자에서 한번 더 inflate하는 코드가 없어질 수 있을 것 같아요. `binding.root`
```suggestion
               DataBindingUtil.inflate(LayoutInflater.from(parent.context), layoutRes, parent, false),
               bindingVariableId
```